packages/next/src/server-build.ts

```diff
@@ -70,6 +70,30 @@ const NEXT_DATA_MIDDLEWARE_RESOLVING_VERSION = 'v12.1.7-canary.33';
 const EMPTY_ALLOW_QUERY_FOR_PRERENDERED_VERSION = 'v12.2.0';
 const CORRECTED_MANIFESTS_VERSION = 'v12.2.0';
 
+// Ideally this should be in a Next.js manifest so we can change it in
+// the future but this also allows us to improve existing versions
```

Comment:
Agreed, this should be in Next.js manifest that we can emit during `next build`. Otherwise it will be painful mapping each version of Next.js each time it needs to change.